src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker_Patterns.cs

```diff
@@ -111,8 +111,26 @@ public override BoundNode VisitNegatedPattern(BoundNegatedPattern node)
 
         public override BoundNode VisitBinaryPattern(BoundBinaryPattern node)
         {
-            Visit(node.Left);
-            Visit(node.Right);
+            // Users (such as ourselves) can have many, many nested binary patterns. To avoid crashing, do left recursion manually.
+
+            var stack = ArrayBuilder<BoundBinaryPattern>.GetInstance();
+            BoundBinaryPattern current = node;
+            do
+            {
+                TakeIncrementalSnapshot(current);
```

Comment:
>TakeIncrementalSnapshot(current); [](http://example.com/codeflow?start=16&length=33)

Is state of ```_expressionIsRead``` insignificant here?

Reply:
It shouldn't be to BinaryPattern itself. If any nested patterns care, they'll save and reset as appropriate, but we don't need to adjust it ourselves here.